alert-common/src/main/java/com/synopsys/integration/alert/common/persistence/model/ConfigurationModel.java

```diff
@@ -88,20 +91,7 @@ public ConfigContextEnum getDescriptorContext() {
         return new HashMap<>(configuredFields);
     }
 
-    public void put(ConfigurationFieldModel configFieldModel) {
-        Objects.requireNonNull(configFieldModel);
-        String fieldKey = configFieldModel.getFieldKey();
-        Objects.requireNonNull(fieldKey);
-        if (configuredFields.containsKey(fieldKey)) {
-            ConfigurationFieldModel oldConfigField = configuredFields.get(fieldKey);
-            List<String> values = combine(oldConfigField, configFieldModel);
-            oldConfigField.setFieldValues(values);
-        } else {
-            configuredFields.put(fieldKey, configFieldModel);
-        }
-    }
-
-    private List<String> combine(ConfigurationFieldModel first, ConfigurationFieldModel second) {
-        return Stream.concat(first.getFieldValues().stream(), second.getFieldValues().stream()).collect(Collectors.toList());
+    public ConfigurationModelMutable createMutableCopy() {
+        return new ConfigurationModelMutable(descriptorId, configurationId, createdAt, lastUpdated, context);
```

Comment:
We also need the `configuredFields` from `this` set on the new instance. You could do this through the constructor or you could do something like this: 
```java
public ConfigurationModelMutable createMutableCopy() {
    ConfigurationModelMutable mutableCopy = new ConfigurationModelMutable(descriptorId, configurationId, createdAt, lastUpdated, context);
    mutableCopy.getConfiguredFields().putAll(configuredFields);
    return mutableCopy;
}
``` 
or (not 100% sure this will work)
```java
public ConfigurationModelMutable createMutableCopy() {
    ConfigurationModelMutable mutableCopy = new ConfigurationModelMutable(descriptorId, configurationId, createdAt, lastUpdated, context);
    mutableCopy.configuredFields = configuredFields;
    return mutableCopy;
}
```

Reply:
I like the putAll approach better.  The equals operator would assign the mutable to the same object reference as the immutable instance.